packages/base/server.js

```diff
@@ -48,107 +48,103 @@ module.exports.run = (executableSchema, middlewares, t, createGraphqlContext) =>
+      next()
     })
+  }
 
-    if (CORS_WHITELIST_URL) {
-      const corsOptions = {
-        origin: CORS_WHITELIST_URL.split(','),
-        credentials: true,
-        optionsSuccessStatus: 200 // some legacy browsers (IE11, various SmartTVs) choke on 204
+  // add req._log()
+  server.use(requestLog)
+
+  // monitor timeouts
+  if (REQ_TIMEOUT) {
+    server.use(
+      timeout(REQ_TIMEOUT, { respond: false }),
+      (req, res, next) => {
+        req.on('timeout', () => {
+          console.log('request timedout:', req._log())
+        })
+        next()
       }
-      server.use('*', cors(corsOptions))
-    }
+    )
+  }
 
-    if (executableSchema) {
-      const graphql = require('./express/graphql')
-      subscriptionServer = graphql(server, pgdb, httpServer, executableSchema, createGraphqlContext)
-    }
+  // Once DB is available, setup sessions and routes for authentication
+  auth.configure({
+    server: server,
+    secret: SESSION_SECRET,
+    domain: COOKIE_DOMAIN || undefined,
+    cookieName: COOKIE_NAME,
+    dev: DEV,
+    pgdb: pgdb
+  })
 
-    for (let middleware of middlewares) {
-      await middleware(server, pgdb, t)
+  if (CORS_WHITELIST_URL) {
+    const corsOptions = {
+      origin: CORS_WHITELIST_URL.split(','),
+      credentials: true,
+      optionsSuccessStatus: 200 // some legacy browsers (IE11, various SmartTVs) choke on 204
     }
+    server.use('*', cors(corsOptions))
+  }
 
-    // start the server
-    if (engine) {
-      engine.listen({
-        port: PORT,
-        httpServer,
-        graphqlPaths: ['/graphql'],
-        launcherOptions: {
-          startupTimeout: 3000
-        }
-      }, () => {
-        console.info('server is running on http://localhost:' + PORT)
-      })
-    } else {
-      httpServer.listen(PORT, () => {
-        console.info('server is running on http://localhost:' + PORT)
-      })
-    }
+  if (executableSchema) {
+    const graphql = require('./express/graphql')
+    subscriptionServer = graphql(server, pgdb, httpServer, executableSchema, createGraphqlContext)
+  }
 
-    return { pgdb }
-  })
+  for (let middleware of middlewares) {
+    await middleware(server, pgdb, t)
+  }
+
+  const callback = () => {
+    console.info('server is running on http://localhost:' + PORT)
+  }
+
+  if (engine) {
+    return engine.listen({
+      port: PORT,
+      httpServer,
+      graphqlPaths: ['/graphql'],
+      launcherOptions: {
+        startupTimeout: 3000
+      }
+    }, callback)
+  } else {
+    return httpServer.listen(PORT, callback)
+  }
 }
 
-module.exports.close = () => {
+const close = () => {
   const { pubsub } = require('./lib/RedisPubSub')
   pubsub.getSubscriber().quit()
   pubsub.getPublisher().quit()
-  if (subscriptionServer) {
-    subscriptionServer.close()
-  }
-  httpServer.close()
-  pgdb.close()
+  subscriptionServer && subscriptionServer.close()
+  httpServer && httpServer.close()
+  pgdb && pgdb.close()
```

Comment:
`pgdb.close()`  returns a promise. Would wait for that one to.

Reply:
I'm against awaiting in close, because it would leave a false impression. The close call comes from a os interrupt, which is asynchronous in it's nature and doesn't care what we do on our end. In the end the process gets killed no matter if we finished closing or not.